Bluepill-cli/bp_test_dumper/bp_test_dumper.m

```diff
@@ -0,0 +1,90 @@
+    return simpleTestName;
+}
+
+/* Return code:
+ * 0 - Successful
+ * 1 - Error loading the test bundles
+ * 2 - Error writing output file
+ */
+__attribute__((constructor))
+void read_it() {
+    NSString *outputFilePath = [[NSProcessInfo processInfo] environment][@"_BP_TEST_LIST_FILE"] ?: @"/tmp/test_list.json";
+    NSLog(@"[BPTestDumper] Output test dump file path is %@", outputFilePath);
+    NSBundle *mainBundle = [NSBundle mainBundle];
+    NSString *pluginsPath = [mainBundle builtInPlugInsPath];
+    NSArray *xcTests = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:pluginsPath error:Nil];
+    NSLog(@"[BPTestDumper] Got a total of %lu xctest bundles.", [xcTests count]);
+    for (NSString *xcTestBundlePath in xcTests) {
+        NSString *ext = [xcTestBundlePath pathExtension];
+        if (![ext isEqual:@"xctest"]) {
+            NSLog(@"[BPTestDumper] Skipping %@ - %@", xcTestBundlePath, ext);
+            continue;
+        }
+        NSString *path = [pluginsPath stringByAppendingPathComponent:xcTestBundlePath];
+        NSLog(@"[BPTestDumper] Loading %@", path);
+        NSBundle *bundle = [NSBundle bundleWithPath:path];
+        if (!bundle) {
+            printf("[BPTestDumper] bundle failed to load!\n");
+            perror("bundle");
+            exit(1);
+        }
+        [bundle load];
+    }
+    NSLog(@"[BPTestDumper] Creating a default test suite with all suites of suites...");
+    XCTestSuite *defaultTestSuite = [XCTestSuite defaultTestSuite];
+    if (!defaultTestSuite) {
+        printf("[BPTestDumper] Failed to create a defaultTestSuite\n");
+        exit(1);
+    }
+    NSLog(@"[BPTestDumper] Successfully created a defaultTestSuite!!!");
+    NSMutableDictionary *testInfoDict = [[NSMutableDictionary alloc] init];
+    for (XCTestSuite *testSuite in [defaultTestSuite tests]) {
+        NSLog(@"[BPTestDumper] Parsing test bundle - %@; Number of test classes = %lu", [testSuite name], [testSuite testCaseCount]);
+        NSMutableDictionary *testClassesDict = [[NSMutableDictionary alloc] init];
+        for (XCTestSuite *testClass in [testSuite tests]) {
+            NSMutableArray *testCaseList = [[NSMutableArray alloc] init];
+            NSLog(@"[BPTestDumper] Parsing test class - %@; Number of test cases = %lu", [testClass name], [testClass testCaseCount]);
+            for (XCTest *testCase in [testClass tests]) {
+                NSLog(@"[BPTestDumper] Test case: %@", getSimpleName([testCase name]));
+                [testCaseList addObject:getSimpleName([testCase name])];
+            }
+            if ([testCaseList count] > 0) {
+                [testClassesDict setObject:(NSArray *)testCaseList forKey:[testClass name]];
+            }
+        }
+        if ([testClassesDict count] > 0) {
+            [testInfoDict setValue:testClassesDict forKey:[testSuite name]];
+        }
+    }
+    assert([NSJSONSerialization isValidJSONObject:testInfoDict]);
```

Comment:
Why would it not be `validJSONObject`, also, do you really want an `assert` here?

Reply:
We don't need the assert here. Also, it is not a proper way to handle errors.